src/backends/simple.ts

```diff
@@ -0,0 +1,442 @@
+     * value is null
+     */
+    previousTargetPTS: null | number = null;
+
+    /**
+     * The number of threads to use for decoding
+     */
+    threadCount = 8;
+
+    /**
+     * The index of the video stream in the demuxer
+     * @param args
+     */
+    streamIndex = 0;
+
+    static async create(args: ExtractorArgs): Promise<Extractor> {
+        const extractor = new SimpleExtractor();
+        await extractor.init(args);
+        return extractor as unknown as Extractor;
+    }
+
+    async init({
+        inputFileOrUrl,
+        threadCount = 8,
+    }: ExtractorArgs): Promise<void> {
+        this.threadCount = threadCount;
+        if (inputFileOrUrl.startsWith('http')) {
+            LOG && console.log('downloading url', inputFileOrUrl);
+            const downloadUrl = new DownloadVideoURL(inputFileOrUrl);
+            await downloadUrl.download();
+            inputFileOrUrl = downloadUrl.filepath;
+            LOG && console.log('finished downloading');
+        }
+        LOG && console.log('loading', inputFileOrUrl);
+        this.demuxer = await beamcoder.demuxer('file:' + inputFileOrUrl);
+        LOG && console.log('streams', this.demuxer.streams.map(stream => stream.codecpar.codec_type));
+        this.streamIndex = this.demuxer.streams.findIndex(stream => stream.codecpar.codec_type === STREAM_TYPE_VIDEO);
+        LOG && console.log('streamIndex', this.streamIndex);
+        if (this.streamIndex === -1) {
+            throw new Error(`File has no ${STREAM_TYPE_VIDEO} stream!`);
+        }
+        LOG && console.log('stream type', this.demuxer.streams[this.streamIndex].codecpar.codec_type);
+        this.filterer = await createFilter({
+            stream: this.demuxer.streams[this.streamIndex],
+            outputPixelFormat: 'rgba',
+        });
+    }
+
+    _createDecoder() {
+        if (this.decoder) {
+            this.decoder.flush();
+            this.decoder = null;
+        }
+        this.decoder = createDecoder({
+            demuxer: this.demuxer as Demuxer,
+            streamIndex: this.streamIndex,
+            threadCount: this.threadCount,
+        });
+    }
+
+    /**
+     * Duration in seconds
+     */
+    get duration(): number {
+        const time_base = this.demuxer.streams[this.streamIndex].time_base;
+        const durations = this.demuxer.streams.map(
+            stream => stream.duration * time_base[0] / time_base[1]
+        );
+
+        return Math.max(...durations);
+    }
+
+    /**
+     * Width in pixels
+     */
+    get width(): number {
+        return this.demuxer.streams[this.streamIndex].codecpar.width;
+    }
+
+    /**
+     * Height in pixels
+     */
+    get height(): number {
+        return this.demuxer.streams[this.streamIndex].codecpar.height;
+    }
+
+    /**
+     * Get the frame for a given time in seconds
+     * @param targetTime
+     */
+    async getFrameAtTime(targetTime: number): Promise<beamcoder.Frame> {
```

Comment:
I think we should use this method more in tests, since we can read `frame.PTS` and ensure it is the expected value. Image tests are easier to mess up.

Reply:
I disagree, it's only when we use an image test that we are sure what the frame contains. That has been crucial for me to find subtle problems. Some examples:
- 2 would render 3
- 13 and 14 would render 14
- the last frames wouldn't render properly
It's only by seeing the actual result that I realized that I had the wrong PTS.

Reply:
@animanathome Valid point, but can you see my next comment then, I think in the best case we check both the PTS and have an image test. (expect(pts).to.be(...) + expect(the image snapshot to match))

Reply:
I don't see the value since we have in test image comparison. If the image is different, the test will fail (and we'll be able to see WHAT is different).